### DISCUSSION TITLE
extend Router during runtime?

### DISCUSSION THREAD

**Discussion by gianpesto** [nuxt/nuxt#7941] [General]
Hi,
i am working on a API driven project and trying to do this with nuxt in SPA mode.
i am using nuxt-auth and nuxt i18n as the site uses Auth0 authorization and is localized by route parameter.

i stumbled upon a problem that i can't wrap my head around, consider the following scenario:

1) User logs in with credentials
2) after being logged in fetch the sitemap from the Server (can change based on user permissions and locale) and looks more or less like this (breaked down):

`"en-us/meow/whatever" :
 {
"pageType": "cat-index" 
}`

so what i want to achieve here is: when the user navigates to the given route, a controller should watch what pagetype is associated to it and return the right page.

My solution until now has been to have a dynamic route that catches all depths, then look up in a map if it finds the path, check the pagetype and import the right page putting it in a `<component>`.

So basically a page that works like a controller and imports all other pages.

It is working, but it doesnt feel right. Does anyone out there have a neater solution to this?

**Comment by marekalgoud** [comment]
Hello,

I'm facing the same kind of issue.
If I do what you explain, we lost the page functionnalities (for me the asyncData to fetch data on server side). I'm looking for a prettier solution too :)

**Comment by gianpesto** [comment]
@marekalgoud yeah i am still having all pages inside the /pages folder which makes them be considered by the router thus have all page-properties.

i accidently stumbled upon this:
https://nuxtjs.org/guide/runtime-config
which seems to be available since 2.13, i still have to try that out but it's worth a shot.

**Comment by gianpesto** [comment]
forget it, the runtime-config only allows you to have env-vars to be exposed to the client

**Comment by gianpesto** [comment]
apparently this will be possibl ein near future:
http://nuxtjs.slides.com/atinux/state-of-nuxt-2020#/5/6/6
